database/slashing.go

```diff
@@ -18,15 +18,19 @@ INSERT INTO validator_signing_info
     (validator_address, start_height, index_offset, jailed_until, tombstoned, missed_blocks_counter, height)
 VALUES `
 	var args []interface{}
+	var counter int
 
-	for i, info := range infos {
-		ii := i * 7
+	for _, info := range infos {
+		if info.ValidatorAddress != "" {
```

Comment:
Instead of wrapping all the logic inside this `if`, I would prefer if we do the other way around: we check if the condition is valid, and we continue the loop if that's the case (skipping the iteration): 

```go
for _, info := range infos {
  if info.ValidatorAddress == "" {
    continue
  }
  
  // Insert the data here
}
```

This would end up in a much more readable code overall